LoopKit/HealthKitSampleStore.swift

```diff
@@ -254,22 +306,37 @@ extension HealthKitSampleStore {
             return
         }
 
-        observerQuery = HKObserverQuery(sampleType: sampleType, predicate: predicate) { [weak self] (query, completionHandler, error) in
-            self?.observeUpdates(to: query, error: error)
+        createObserverQuery(predicate: predicate)
+    }
 
-            completionHandler()
+    private func createObserverQuery(predicate: NSPredicate) {
+        observerQuery = HKObserverQuery(sampleType: sampleType, predicate: predicate) { [weak self] (query, completionHandler, error) in
+            self?.observeUpdates(to: query, completionHandler: completionHandler, error: error)
         }
 
         enableBackgroundDelivery { (result) in
             switch result {
             case .failure(let error):
                 self.log.error("Error enabling background delivery: %@", error.localizedDescription)
             case .success:
-                self.log.debug("Enabled background delivery for %@", self.sampleType)
+                self.log.default("Enabled background delivery for %{public}@", self.sampleType)
             }
         }
     }
 
+    private func createAnchoredObjectQuery(predicate: NSPredicate?, anchor: HKQueryAnchor?) {
+        let anchoredObjectQuery = HKAnchoredObjectQuery(
+            type: sampleType,
+            predicate: anchor == nil ? predicate : nil,
```

Comment:
The predicate in this case is limiting the query to `endDate >= observationStart`, which is mainly applicable to the first query when anchor is nil. I doubt it costs much to keep that predicate in effect for subsequent queries, and it would probably be more of an expected behavior if this method was re-used for other purposes, so I"m ok adding the predicate back in, even when we do have an anchor.

Reply:
I agree with @darinkrauss .  Sounds like you do too, @ps2 ?

Reply:
Actually, that specific predicate does protect against someone adding an HKObject in the past (earlier than `observationStart` at least) via a third-party app. Granted, that's not likely, but still.